src/relentless/optimize/objective.py

```diff
@@ -353,15 +352,25 @@ def compute(self, variables, directory=None):
         # run simulation and use result to compute gradient
         try:
             sim = self.simulation.run(self.potentials, directory)
-            sim_ens = sim[self.thermo]["ensemble"]
+            if self._use_ensemble_average(self.target, self.thermo):
+                sim_ens = sim.directory.file(self.thermo.filename)
```

Comment:
Error check needed somewhere to ensure this is a GSD file? Otherwise I think the calculation method will choke when it tries to open it.